pkg/controller/kieapp/defaults/defaults_test.go

```diff
@@ -6805,3 +6805,141 @@ func TestDashbuilderWithPartialCORS(t *testing.T) {
+}
+
+func TestKieServerControllerStartupStrategyConfiguration(t *testing.T) {
+	cr := &api.KieApp{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: "test",
+		},
+		Spec: api.KieAppSpec{
+			Environment: api.RhpamProductionImmutable,
+			CommonConfig: api.CommonConfig{
+				StartupStrategy: createStartupStrategy(api.ControllerStartupStrategy, 0),
+			},
+			Objects: api.KieAppObjects{
+				Servers: []api.KieServerSet{},
+			},
+		},
+	}
+
+	env, err := GetEnvironment(cr, test.MockService())
+	assert.Nil(t, err, "Error getting prod environment")
+	assert.NotNil(t, env)
+	assert.Equal(t, api.ControllerStartupStrategy, getEnvVariable(env.Servers[0].DeploymentConfigs[0].Spec.Template.Spec.Containers[0], "KIE_SERVER_STARTUP_STRATEGY"))
+	assert.Equal(t, "false", getEnvVariable(env.Servers[0].DeploymentConfigs[0].Spec.Template.Spec.Containers[0], "KIE_SERVER_CONTROLLER_OPENSHIFT_ENABLED"))
+	assert.Equal(t, "", getEnvVariable(env.Servers[0].DeploymentConfigs[0].Spec.Template.Spec.Containers[0], "KIE_SERVER_CONTROLLER_TEMPLATE_CACHE_TTL"))
+}
+
+func checkConsoleControllerStrategyAssertions(t *testing.T, cr *api.KieApp) {
+	env, err := GetEnvironment(cr, test.MockService())
+	assert.Nil(t, err, "Error getting prod environment")
+	assert.NotNil(t, env)
+	assert.Equal(t, "false", getEnvVariable(env.Console.DeploymentConfigs[0].Spec.Template.Spec.Containers[0], "KIE_SERVER_CONTROLLER_OPENSHIFT_ENABLED"))
+}
+
+func checkConsoleOpenshiftStrategyAssertions(t *testing.T, cr *api.KieApp, assertion bool, ttl string) {
+	env, err := GetEnvironment(cr, test.MockService())
+	assert.Nil(t, err, "Error getting prod environment")
+	assert.NotNil(t, env)
+
+	assert.Equal(t, "true", getEnvVariable(env.Console.DeploymentConfigs[0].Spec.Template.Spec.Containers[0], "KIE_SERVER_CONTROLLER_OPENSHIFT_ENABLED"))
+	assert.Equal(t, "true", getEnvVariable(env.Console.DeploymentConfigs[0].Spec.Template.Spec.Containers[0], "KIE_SERVER_CONTROLLER_OPENSHIFT_GLOBAL_DISCOVERY_ENABLED"))
+	assert.Equal(t, "true", getEnvVariable(env.Console.DeploymentConfigs[0].Spec.Template.Spec.Containers[0], "KIE_SERVER_CONTROLLER_OPENSHIFT_PREFER_KIESERVER_SERVICE"))
+	assert.Equal(t, ttl, getEnvVariable(env.Console.DeploymentConfigs[0].Spec.Template.Spec.Containers[0], "KIE_SERVER_CONTROLLER_TEMPLATE_CACHE_TTL"))
+}
+
+func TestConsoleOpenshitStartupStrategyConfiguration(t *testing.T) {
+	cr := &api.KieApp{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: "test",
+		},
+		Spec: api.KieAppSpec{
+			Environment: api.RhpamAuthoring,
+			CommonConfig: api.CommonConfig{
+				StartupStrategy: createStartupStrategy(api.OpenshiftStartupStrategy, 4000),
+			},
+			Objects: api.KieAppObjects{
+				Console: &api.ConsoleObject{},
+			},
+		},
+	}
+
+	checkConsoleOpenshiftStrategyAssertions(t, cr, true, "4000")
+}
+
+func TestConsoleControllerStartupStrategyConfiguration(t *testing.T) {
```

Comment:
please merge this with TestKieServerControllerStartupStrategyConfiguration

Reply:
ok